stdlib/array.ncl

```diff
@@ -128,10 +133,12 @@
         ```
         "%
       = fun f fst l =>
-        if %length% l == 0 then
-          fst
-        else
-          f (%head% l) (fold f fst (%tail% l)),
+        let rec go = fun n =>
+          if n == %length% l then
```

Comment:
I don't think it'll necessarily make a huge difference overall but I believe adding a `let length = %length% l in` before defining `go` here will be marginally more performant than repeatedly calling the `%length%` primop in the loop.

Reply:
Turns out the difference isn't really measurable within 1% or so. But I like having it outside on principle 👍